### ISSUE TITLE
deploy_bosh.sh fails when creating stemcell in get_storage_account_keys_byname

### ISSUE THREAD

**Issue by mdcarlson** [cloudfoundry-incubator/bosh-azure-cpi-release#94]
I created a new trial account and used the deployment template to create a VM for the bosh installation.

Bosh deploy fails with the following error when attempting to create the stemcell.

    CPI 'create_stemcell' method responded with error: CmdError{"type":"Bosh::Clouds::CloudError","message":"http_get_response - retry 0: #\u003cBosh::AzureCloud::AzureError: get_token - http error: 400\u003e
    /home/bosh/.bosh_init/installations/8c931220-e6a5-46b0-507f-b360dd201005/packages/bosh_azure_cpi/lib/cloud/azure/azure_client2.rb:848:in `get_token'
    /home/bosh/.bosh_init/installations/8c931220-e6a5-46b0-507f-b360dd201005/packages/bosh_azure_cpi/lib/cloud/azure/azure_client2.rb:869:in `http_get_response'
    /home/bosh/.bosh_init/installations/8c931220-e6a5-46b0-507f-b360dd201005/packages/bosh_azure_cpi/lib/cloud/azure/azure_client2.rb:1035:in `http_post'
    /home/bosh/.bosh_init/installations/8c931220-e6a5-46b0-507f-b360dd201005/packages/bosh_azure_cpi/lib/cloud/azure/azure_client2.rb:795:in `get_storage_account_keys_by_name'
    /home/bosh/.bosh_init/installations/8c931220-e6a5-46b0-507f-b360dd201005/packages/bosh_azure_cpi/lib/cloud/azure/table_manager.rb:13:in `initialize'

This is a new Azure account and trial subscription.  There does appear to be a valid storage account that contains some items in the `vids` storage container.

Gist for last 1000 lines of run.log:  https://gist.github.com/mdcarlson/57633c50c436bafc2bb2 

I have deleted the deployment and retried a fresh install with the same results.

Regards,

Mark

**Comment by AbelHu** [comment]
It seems like that you have skipped an important step. It is to create a work or school account.
You cannot use the trial account directly. You should use it to create an organization account. Please reference this [link] (https://azure.microsoft.com/en-us/documentation/articles/resource-group-create-work-id-from-personal/) and then use the new account.

**Comment by svrc-pivotal** [comment]
We were seeing this problem in the Azure workshop room yesterday as well with one user, will need to verify the missing step today.

**Comment by mdcarlson** [comment]
@AbelHu:  Thanks for the reply.

I created a new account following the instructions at the link and used it for the template-based deploy.

I had to explicitly logout from the initial free trial account to ensure that my service principal was created with the subscription for the new AD user.

It might be helpful to add a caution about this step in the instructions [here](https://github.com/cloudfoundry-incubator/bosh-azure-cpi-release/blob/master/docs/guidance.md).
